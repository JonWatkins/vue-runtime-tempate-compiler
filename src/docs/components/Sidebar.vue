<template>
  <div class="col-md-3 col-xl-3 position-fixed sidebar">
    <nav>
      <ul class="nav flex-column">
        <li class="nav-item">
          <router-link to="/">Introduction</router-link>
        </li>
        <li class="nav-item">
          <router-link to="/installation">Installation</router-link>
        </li>
        <li class="nav-item">
          <router-link to="/guide/getting-started">Getting Started</router-link>
        </li>
        <li class="nav-item">
          <router-link to="/guide/parent-context">Parent Context</router-link>
        </li>
        <li class="nav-item">
          <router-link to="/guide/methods">Methods</router-link>
        </li>
        <li class="nav-item">
          <router-link to="/guide/computed">Computed</router-link>
        </li>
        <li class="nav-item">
          <router-link to="/guide/props">Props</router-link>
        </li>
        <li class="nav-item">
          <router-link to="/guide/provide">Provide</router-link>
        </li>
        <li class="nav-item">
          <router-link to="/guide/filters">Filters</router-link>
        </li>
      </ul>
    </nav>
  </div>
</template>

<script>
export default {
  name: 'Sidebar'
}
</script>

<style lang="scss" rel="stylesheet/scss">
.sidebar {
  position: sticky;
  z-index: 1000;
  height: 100vh;
  border-right: 1px solid #eaecef;
  padding: 1.5rem 0;

  a {
    font-weight: 400;
    display: inline-block;
    color: #2c3e50;
    border-left: 0.25rem solid transparent;
    padding: 0.35rem 1rem 0.35rem 1.25rem;
    line-height: 1.4;
    width: 100%;
    box-sizing: border-box;

    &.active-exact,
    &:hover {
      color: #3eaf7c;
    }

    &.active-exact {
      border-left-color: #3eaf7c;
    }
  }
}
</style>
