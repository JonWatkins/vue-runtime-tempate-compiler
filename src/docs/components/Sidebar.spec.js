import Sidebar from './Sidebar.vue'
const { shallowMountLocal } = global

let wrapper

describe('Sidebar', () => {
  beforeEach(() => {
    wrapper = shallowMountLocal(Sidebar)
  })

  it('is called', () => {
    expect(wrapper.name()).toBe('Sidebar')
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
