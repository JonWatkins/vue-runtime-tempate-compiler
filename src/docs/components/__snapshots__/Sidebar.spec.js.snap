// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar render correctly 1`] = `
<div class="col-md-3 col-xl-3 position-fixed sidebar d-none d-sm-none d-md-block">
  <nav>
    <ul class="nav flex-column">
      <li class="nav-item">
        <router-link-stub to="/" tag="a" event="click">Introduction</router-link-stub>
      </li>
      <li class="nav-item">
        <router-link-stub to="/installation" tag="a" event="click">Installation</router-link-stub>
      </li>
      <li class="nav-item">
        <router-link-stub to="/guide/getting-started" tag="a" event="click">Getting Started</router-link-stub>
      </li>
      <li class="nav-item">
        <router-link-stub to="/guide/parent-context" tag="a" event="click">Parent Context</router-link-stub>
      </li>
      <li class="nav-item">
        <router-link-stub to="/guide/methods" tag="a" event="click">Methods</router-link-stub>
      </li>
      <li class="nav-item">
        <router-link-stub to="/guide/computed" tag="a" event="click">Computed</router-link-stub>
      </li>
      <li class="nav-item">
        <router-link-stub to="/guide/props" tag="a" event="click">Props</router-link-stub>
      </li>
      <li class="nav-item">
        <router-link-stub to="/guide/provide" tag="a" event="click">Provide</router-link-stub>
      </li>
      <li class="nav-item">
        <router-link-stub to="/guide/filters" tag="a" event="click">Filters</router-link-stub>
      </li>
    </ul>
  </nav>
</div>
`;
